kuttl-tests/Makefile

```diff
@@ -11,44 +12,55 @@ endif
 
-bin/kubectl_$(KUBERNETES_VERSION):
+bin/kubectl_$(KUBERNETES_VERSION)_$(OS):
 	mkdir -p bin/
-	curl -Lo bin/kubectl_$(KUBERNETES_VERSION) https://storage.googleapis.com/kubernetes-release/release/v$(KUBERNETES_VERSION)/bin/$(OS)/$(MACHINE)/kubectl
-	chmod +x bin/kubectl_$(KUBERNETES_VERSION)
+	curl -Lo bin/kubectl_$(KUBERNETES_VERSION)_$(OS) https://storage.googleapis.com/kubernetes-release/release/v$(KUBERNETES_VERSION)/bin/$(OS)/$(MACHINE)/kubectl
+	chmod +x bin/kubectl_$(KUBERNETES_VERSION)_$(OS)
 
-bin/kubectl: bin/kubectl_$(KUBERNETES_VERSION)
-	ln -sf ./kubectl_$(KUBERNETES_VERSION) ./bin/kubectl
+bin/kubectl: bin/kubectl_$(KUBERNETES_VERSION)_$(OS)
+	ln -sf ./kubectl_$(KUBERNETES_VERSION)_$(OS) ./bin/kubectl
 
-bin/kubectl-kuttl_$(KUTTL_VERSION):
+bin/kubectl-kuttl_$(KUTTL_VERSION)_$(OS):
 	mkdir -p bin/
-	curl -Lo bin/kubectl-kuttl_$(KUTTL_VERSION) https://github.com/kudobuilder/kuttl/releases/download/v$(KUTTL_VERSION)/kubectl-kuttl_$(KUTTL_VERSION)_$(OS)_$(KUDO_MACHINE)
-	chmod +x bin/kubectl-kuttl_$(KUTTL_VERSION)
+	curl -Lo bin/kubectl-kuttl_$(KUTTL_VERSION)_$(OS) https://github.com/kudobuilder/kuttl/releases/download/v$(KUTTL_VERSION)/kubectl-kuttl_$(KUTTL_VERSION)_$(OS)_$(KUDO_MACHINE)
+	chmod +x bin/kubectl-kuttl_$(KUTTL_VERSION)_$(OS)
 
-bin/kubectl-kuttl: bin/kubectl-kuttl_$(KUTTL_VERSION)
-	ln -sf ./kubectl-kuttl_$(KUTTL_VERSION) ./bin/kubectl-kuttl
+bin/kubectl-kuttl: bin/kubectl-kuttl_$(KUTTL_VERSION)_$(OS)
+	echo "Linking kubectl-kuttl_$(KUTTL_VERSION)_$(OS)"
+	ln -sf ./kubectl-kuttl_$(KUTTL_VERSION)_$(OS) ./bin/kubectl-kuttl
 
-bin/kubectl-kudo_$(KUDO_VERSION):
+bin/kubectl-kudo_$(KUDO_VERSION)_$(OS):
 	mkdir -p bin/
-	curl -Lo bin/kubectl-kudo_$(KUDO_VERSION) https://github.com/kudobuilder/kudo/releases/download/v$(KUDO_VERSION)/kubectl-kudo_$(KUDO_VERSION)_$(OS)_$(KUDO_MACHINE)
-	chmod +x bin/kubectl-kudo_$(KUDO_VERSION)
+	curl -Lo bin/kubectl-kudo_$(KUDO_VERSION)_$(OS) https://github.com/kudobuilder/kudo/releases/download/v$(KUDO_VERSION)/kubectl-kudo_$(KUDO_VERSION)_$(OS)_$(KUDO_MACHINE)
+	chmod +x bin/kubectl-kudo_$(KUDO_VERSION)_$(OS)
 
-bin/kubectl-kudo: bin/kubectl-kudo_$(KUDO_VERSION)
-	ln -sf ./kubectl-kudo_$(KUDO_VERSION) ./bin/kubectl-kudo
+bin/kubectl-kudo: bin/kubectl-kudo_$(KUDO_VERSION)_$(OS)
+	ln -sf ./kubectl-kudo_$(KUDO_VERSION)_$(OS) ./bin/kubectl-kudo
+
+bin/kind_$(KIND_VERSION)_$(OS):
+	mkdir -p bin/
+	curl -Lo bin/kind_$(KIND_VERSION)_$(OS) https://github.com/kubernetes-sigs/kind/releases/download/v$(KIND_VERSION)/kind-$(OS)-$(MACHINE)
+	chmod +x bin/kind_$(KIND_VERSION)_$(OS)
+
+bin/kind: bin/kind_$(KIND_VERSION)_$(OS)
+	ln -sf ./kind_$(KIND_VERSION)_$(OS) ./bin/kind
 
 .PHONY: install-kuttl
-install-kuttl: bin/kubectl-kuttl bin/kubectl-kudo bin/kubectl
+install-kuttl: bin/kubectl-kuttl bin/kubectl-kudo bin/kubectl bin/kind
 
 .PHONY: kind-test
 kind-test: install-kuttl render-test-templates
 	mkdir -p kuttl-dist/
 	go get github.com/jstemmer/go-junit-report
+	kind delete cluster
```

Comment:
There seem to be some cases where the kind cluster continues to exist over builds - I think we don't get a new build node each time, they are reused. To be sure, I've added the kind delete cluster here.